packages/types/src/type/Vote.ts

```diff
@@ -2,41 +2,79 @@
 // This software may be modified and distributed under the terms
 // of the Apache-2.0 license. See the LICENSE file for details.
 
-import { isBoolean } from '@polkadot/util';
+import { isBoolean, isNumber, isObject, u8aToBn, isU8a, bnToU8a } from '@polkadot/util';
 
-import I8 from '../primitive/I8';
+import Conviction from './Conviction';
+import U8a from '../codec/U8a';
+import Boolean from '../primitive/Bool';
 
 /**
  * @name Vote
  * @description
  * A number of lock periods, plus a vote, one way or the other.
  */
-export default class Vote extends I8 {
+export default class Vote extends U8a {
+  private _aye: Boolean;
+  private _conviction?: Conviction;
+
   constructor (value?: any) {
-    super(Vote.decodeVote(value));
+    // decoded is just 1 byte
+    // Aye: Most Significant Bit
+    // Conviction: 0000 - 0101
+    const decoded = Vote.decodeVote(value);
+
+    super(decoded);
+
+    const msb = decoded[0] << 7;
+    const conviction = decoded[0] & ((1 << 2) - 1);
+
+    this._aye = new Boolean(msb);
+    this._conviction = new Conviction(conviction);
   }
 
-  private static decodeVote (value?: any): any {
+  private static decodeVote (value?: any): Uint8Array {
     if (isBoolean(value)) {
-      return value ? -1 : 0;
+      return value ? new Uint8Array([-1]) : new Uint8Array([0]);
     } else if (value instanceof Boolean) {
       return Vote.decodeVote(value.valueOf());
+    } else if (isNumber(value)) {
+      return value < 0 ? bnToU8a(1, { bitLength: -1, isLe: true, isNegative: true }) : bnToU8a(0);
+    } else if (isObject(value)) {
+      const aye = value.aye;
+      const index = value.conviction.toNumber();
+      const conviction = index >>> 0;
```

Comment:
I've never used `>>>` before, so not 100% sure, but I would say this line does nothing?